src/application/ventilation.c

```diff
@@ -0,0 +1,80 @@
+/*
+ * ventilation.c
+ *
+ */
+
+#include <stdbool.h>
+#include <platform/systemInfo.h>
+#include <platform/pressureFlowSensor.h>
+#include <platform/valve.h>
+#include <application/ventilation.h>
+#include <application/dss.h>
+
+#define INSPIRATORY_FLOW_TIME 	1
+#define INSPIRATORY_PAUSE	2
+#define EXPIRATORY_FLOW_TIME 	3
+#define EXPIRATORY_PAUSE	4
+
+/*
+ * the time the current breath cycle was supposed to start
+ */
+static int start_current_breath_cycle = 0;
+
+/*
+ * the time the current breath cycle was supposed to start
+ */
+static int breath_cycle_duration = 125; //to do replace this calue by the real one
+
+/*
+ * Phase of the ventilation
+ * This should always be one of
+ *  INSPIRATORY_FLOW_TIME | INSPIRATORY_PAUSE
+ *  | EXPIRATORY_FLOW_TIME | EXPIRATORY_PAUSE
+ */
+
+int get_circuit_pressure() { return 0; }
+
+static int ventilation_phase = EXPIRATORY_PAUSE;
+
+void ventilation(){
+  switch (ventilation_phase) {
+    case INSPIRATORY_FLOW_TIME:
+      start_inspiration();
+      break;
+
+    case INSPIRATORY_PAUSE:
+      // statements
+      break;
+
+    case EXPIRATORY_FLOW_TIME:
+      // statements
+      break;
+
+    case EXPIRATORY_PAUSE:
+      // statements
+      break;
+
+    default:
+      dss();
+  }
+}
+
+void start_inspiration(){
+  int current_time = get_current_time();
+  if (ventilation_phase ==  EXPIRATORY_PAUSE

```

Comment:
I am confused by the first part of this conjunction.

This checks that the current phase is the EXPIRATORY_PHASE (ok, that's fine). But then above, this function is only ever called in the INSPIRATORY_FLOW_TIME phase. Is there something wrong here?

Reply:
good catch modified